### ISSUE TITLE
migration run doesn't produce anything on postgres

### ISSUE THREAD

**Issue by nathanielks** [phalcon/phalcon-devtools#290]
Heya!

Just provisioned a new server and am trying to get the database up and running. I'm running `phalcon migration run` and it spits out the tool version, but doesn't do anything else. I've checked the database to see if there's anything there and nothing. Does the database need existing tables in order for it to migrate the versions?

**Comment by nathanielks** [comment]
Perhaps #237 is related?

**Comment by rianorie** [comment]
I noticed the same and fixed it, just haven't gotten around to getting it on here.. give me a few minutes :)

**Comment by nathanielks** [comment]
you're awesome @rianorie =D

**Comment by rianorie** [comment]
I'm behind a few versions it seems, I'm gonna need a little more than a "few" minutes. I need to reproduce what I fixed before. Shouldn't be super long, though.

**Comment by nathanielks** [comment]
okie doke :+1:

**Comment by rianorie** [comment]
@nathanielks there you go. It has to be merged before you can use it though. Optionally use https://github.com/rianorie/phalcon-devtools but you want to do this only temporarily, it's likely my fork will end up lagging behind often.

**Comment by nathanielks** [comment]
I'll give that a try and report back shortly!

**Comment by nathanielks** [comment]
It didn't work :(

**Comment by nathanielks** [comment]
aha! It did this time! But I don't know what it means =\

```
1422311894.0557: SELECT CASE WHEN COUNT(*) > 0 THEN 1 ELSE 0 END FROM information_schema.tables WHERE table_schema = 'db_name' AND table_name='queryerror'  => 1422311894.3119 (0.25621700286865)
                              
  Error: Not implemented yet  

```

**Comment by rianorie** [comment]
I'm not sure, that "Not implemented yet" message isn't part of the devtools. hm..

**Comment by nathanielks** [comment]
It's popped up in #119 too. Context: this is my first phalcon project. Built off of https://github.com/jeteokeeffe/php-hmac-rest-api. I had to make a few adjustments to get it to work with phalcon devtools. `phalcon migration generate` works just fine, but the `run` action on the remote server isn't playing nicely. It's worked fine locally though.

**Comment by rianorie** [comment]
different php versions? mysql versions? phalcon versions?

**Comment by nathanielks** [comment]
Nope, all identical as I provisioned with Ansible. ( postgres, fwiw )

**Comment by rianorie** [comment]
so locally both `generate` and `run` work, but not on the server?

**Comment by nathanielks** [comment]
`generate` does, yup. `run` actually spits out the same errors in both locations.

**Comment by rianorie** [comment]
any chance I can get you to share the migration you're trying to run?

**Comment by nathanielks** [comment]
yup, let me put it together for you.

**Comment by nathanielks** [comment]
http://trp.do/FvVbgwb

That'll be destroyed after the first download, so don't delete it too hastily ;)

**Comment by AlfonsoPerez** [comment]
Dayum, I arrived too late :-)

**Comment by nathanielks** [comment]
HA. I'll repost for you @AlfonsoPerez =p

http://trp.do/OWQEX4w

**Comment by rianorie** [comment]
@nathanielks  your db is called `db_name` ? If it's not.. update your config file (`app/config/config.ini` probably) and try running your migrations again.

I did get a SQL error from the migrations, but overal they look fine. However I'm now noticing something funky in your earlier error message:
`1422311894.0557: SELECT CASE WHEN COUNT(*) > 0 THEN 1 ELSE 0 END FROM information_schema.tables WHERE table_schema = 'db_name' AND table_name='queryerror'  => 1422311894.3119 (0.25621700286865)`

Especially the `table_schema = 'db_name' ` part is interest. That is "where database is called db_name", which seems like an unlikely name for a database :)

**Comment by nathanielks** [comment]
haha, no no, just changed it for googling reasons.

**Comment by rianorie** [comment]
damn, well, in that case I'm sad to say I can't reproduce your problem. I have to add I'm using MySQL rather than Postgres though, so maybe that's messing things up. At which point you're back to the database not liking whatever you're trying to do to it. I'd probably remove all but the api migration and try to get that imported by removing columns and indexes until it works. Then re-add said elements one by one while trying to run the migration.

**Comment by nathanielks** [comment]
Will do @rianorie! Thanks for looking into this!

**Comment by nathanielks** [comment]
Hey @rianorie, here's the issue I do believe for future reference: http://forum.phalconphp.com/discussion/3593/phalcon-postgres-migration-run-error-not-implemented-yet

The features literally haven't been implemented yet.

**Comment by nathanielks** [comment]
( for Postgres )

**Comment by rianorie** [comment]
Well, that won't do at all. I'd be very open to pullrequests from someone with postgres experience. Though I believe it uses mostly the same dialect as MySQL so it may not be that big a hassle really.

Thanks for the headsup!

**Comment by nathanielks** [comment]
Absolutely dude! Unfortunately this is my first time using Postgres too and it was mainly an experiment. Would love to use it in the future though!

**Comment by nathanielks** [comment]
Le sigh. I've switched to Mysql, using your repo, and it's still not migrating on first run :(. It outputs the devtools version and that's it.

Edit: success! I followed [this tip](https://github.com/phalcon/phalcon-devtools/issues/226#issuecomment-47418449) and it worked! Need to update the migration-version and it ran.

**Comment by rianorie** [comment]
Created a new uncluttered endpoint for this, see #295.
